### DISCUSSION TITLE
how to plot partial ecg-data

### DISCUSSION THREAD

**Discussion by an4di** [neuropsychology/NeuroKit#681] [Q&A]
Hey community - I am an undergrad, trying to process ECG data with neurokit2 and need your help!
In order to compare closer the artifacts and peaks of manually cleaned data (mindware, intervalrelated) with neurokit's cleaning methods, I want to plot only an extracted range of datapoints (1min) out of the full processed dataset.
As mentioned [in the docs](https://neuropsychology.github.io/NeuroKit/examples/bio_intervalrelated/bio_intervalrelated.html) I can select any range of datapoints from the start (e.g [:30000]), but, when I try to extract a range from the middle (e.g [30000:60000]) it does not work. I tried many options, including taking the ecg_plot source code apart, but nothing works. I would appreciate any help here and can provide the scripts if needed. Thank you!

**Comment by DominiqueMakowski** [comment]
could you post the code that you tried and the error log?

**Reply by an4di** [reply]
that was the first attempt, which did not work:

import neurokit2 as nk
import pandas as pd
import matplotlib.pyplot as plt

data = pd.read_csv("C:/...",  sep='\t')
ecg, info = nk.ecg_process(data["MWMOBILE_Bio"], sampling_rate=500)
nk.ecg_plot(ecg[60000:90000], sampling_rate=500)




Error:
KeyError: "None of [Int64Index([  ... ],\n           dtype='int64')] are in the [index]"

**Comment by DominiqueMakowski** [comment]
Try 

```python
ecg = ecg.reset_index()
nk.ecg_plot(ecg[0:2000], sampling_rate=500)
```

**Reply by an4di** [reply]
I tried that one too. Worked for [0:2000], but not for other sliced parts (e.g[3000:6000]).  throws the same Error :-(

**Reply by DominiqueMakowski** [reply]
Maybe you don't enough data points in your signal? What's the length `len(ecg)`?

**Reply by an4di** [reply]
len(data)
Out[27]: 1717500


I tried to do the same with my EDA-data, but it throws the same Errors. I believe it is an issue inside the nk.ecg_process / eda_process that does not allow me an easy extraction of partial data.

**Reply by DominiqueMakowski** [reply]
Strange. Try to convert the series to a vector `ecg.values[60000:90000]` and plot that

**Reply by an4di** [reply]
nope, not working. Is it working for you/anybody the "easy" way [start:end]?
We created now our own plot, slicing the data beforehand
